idl/github.com/uber/cadence/history.thrift

```diff
@@ -86,6 +86,31 @@ struct GetMutableStateResponse {
   150: optional shared.VersionHistories versionHistories
 }
 
+struct PollMutableStateRequest {
+  10: optional string domainUUID
+  20: optional shared.WorkflowExecution execution
+  30: optional i64 (js.type = "Long") expectedNextEventId
+}
+
+struct PollMutableStateResponse {
+  10: optional shared.WorkflowExecution execution
+  20: optional shared.WorkflowType workflowType
+  30: optional i64 (js.type = "Long") NextEventId
+  35: optional i64 (js.type = "Long") PreviousStartedEventId
+  40: optional i64 (js.type = "Long") LastFirstEventId
+  50: optional shared.TaskList taskList
+  60: optional shared.TaskList stickyTaskList
+  70: optional string clientLibraryVersion
+  80: optional string clientFeatureVersion
+  90: optional string clientImpl
+  100: optional bool isWorkflowRunning
```

Comment:
plz change this to workflow state
also add a workflow close status

Reply:
Updated. Please take another look